src/openrct2/localisation/Formatting.cpp

```diff
@@ -0,0 +1,207 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2020 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "Formatting.h"
+
+#include "../config/Config.h"
+#include "../util/Util.h"
+#include "FormatCodes.h"
+#include "Language.h"
+#include "StringIds.h"
+
+#include <cstdint>
+
+namespace OpenRCT2
+{
+    char GetDigitSeperator()
+    {
+        return ',';
+    }
+
+    char GetDecimalSeperator()
+    {
+        return '.';
+    }
+
+    template<size_t TDecimalPlace, bool TDigitSep, typename T> void FormatNumber(std::stringstream& ss, T value)
+    {
+        char buffer[32];
+        int32_t i = 0;
+
+        size_t num;
+        if (value < 0)
+        {
+            // TODO handle edge case: std::numeric_limits<int64_t>::min();
+            num = -value;
+            ss << '-';
+        }
+        else
+        {
+            num = value;
+        }
+
+        // Decimal digits
+        if constexpr (TDecimalPlace > 0)
+        {
+            while (num != 0 && i < sizeof(buffer) && i < TDecimalPlace)
+            {
+                buffer[i++] = (char)('0' + (num % 10));
```

Comment:
I guess this probably failed compilation due to the usage of c-style cast (but as I'm looking commit by commit, I won't know now :D)

Reply:
Yeah, I changed to C++ cast.